quinn-proto/src/endpoint.rs

```diff
@@ -232,6 +232,18 @@ impl Endpoint {
                         return None;
                     }
                 };
+
+            if let Err(reason) = self.early_validate_first_packet(dst_cid) {
+                return Some(DatagramEvent::Response(self.initial_close(
```

Comment:
Should/could we have `initial_close()` take `&InitialHeader` instead of separate `version` and `src_cid`?

Reply:
We sometimes call `initial_close` after decrypting the packet, when a plain header is no longer convenient.